standard/PortalStatistics.lua

```diff
@@ -0,0 +1,1138 @@
+	end
+
+	local categoryNames = nonYearCategories
+	local seriesNames = yearList
+
+	if transpose == true then
+		seriesData = Array.map(Array.range(1, Table.size(nonYearCategories)), function(index)
+			return Array.map(seriesData, function(teamData)
+					return teamData[index] or 0
+				end)
+		end)
+		seriesNames, categoryNames = categoryNames, seriesNames
+	end
+
+	if config.removeEmptyCategories == true then
+		local startsEmpty = true
+		local lastNotEmpty = 1
+
+		local isEmptyCategory = Array.map(Array.map(categoryNames, function(_, catIndex)
+				local truthValue = Array.all(Array.map(seriesData, function(_, index)
+					return seriesData[index][catIndex] end), function(value)
+						return value == 0
+					end)
+				if truthValue == false then
+					lastNotEmpty = catIndex
+				end
+				return truthValue
+			end),
+		function(value, index)
+			if index > lastNotEmpty then
+				return false
+			elseif startsEmpty == true and value == true then
+				return false
+			else
+				startsEmpty = false
+				return true
+			end
+		end)
+
+		categoryNames = Array.filter(categoryNames, function(_, catIndex)
+				return Logic.readBool(isEmptyCategory[catIndex]) end)
+
+		seriesData = Array.map(seriesData, function(_, index)
+				return Array.filter(seriesData[index], function(_, catIndex)
+						return Logic.readBool(isEmptyCategory[catIndex]) end)
+			end)
+	end
+
+	for seriesIndex, series in pairs(seriesNames) do
+		if config.removeEmptySeries == true and Array.all(seriesData[seriesIndex], function(value)
+			return value == 0
+		end) then
+			mw.logObject(series .. ' is empty')
+		else
+			table.insert(chartData, {
+					['name'] = series,
+					['type'] = 'bar',
+					['stack'] = 'total',
+					['data'] = seriesData[seriesIndex],
+					['emphasis'] = {['focus'] = 'series'},
+				}
+			)
+		end
+	end
+
+	config['yAxis'] = {type = 'value', name = 'Earnings ($USD)'}
+	config['xAxis'] = {type = 'category', name = config.catLabel, data = categoryNames}
+	config['customLegend'] = config['customLegend'] or seriesNames
+
+	return StatisticsPortal._drawChart(config, chartData)
+end
+
+
+function StatisticsPortal._chartConfig(args, params)
+
+	local isForTeam = String.isNotEmpty(args.team) or Logic.readBool(args.isForTeam)
+
+	return {
+		processFunction = params.processFunction or nil,
+		variable = params.variable or nil,
+		catLabel = params.catLabel,
+		flipAxes = params.flipAxes or false,
+
+		customLegend = String.isNotEmpty(args.customLegend) and
+			Array.map(mw.text.split(args.customLegend, ','), String.trim) or args.customInputs,
+		customInputs = args.customInputs,
+		customYears = args.customYears,
+		startYear = args.startYear or Info.startYear,
+		yearBreakdown = Logic.readBoolOrNil(args.yearBreakdown),
+		removeEmptyCategories = Logic.readBool(args.removeEmptyCategories) or false,
+		removeEmptySeries = Logic.readBool(args.removeEmptySeries) or false,
+		isForTeam = isForTeam,
+		opponentName = isForTeam and args.team or args.player,
```

Comment:
mind that if
- isForTeam is true
- args.team is unset (nil)
- args.player is set

args player will be used here

Reply:
resolved to make it so that if isForTeam is true, then opponentName can only be args.team or nil